src/linker/Linker.Steps/CheckSuppressionsDispatcher.cs

```diff
@@ -23,13 +24,30 @@ public override void Process (LinkContext context)
 			// Suppressions targeting RedundantSuppression warning should not be reported.
 			redundantSuppressions = redundantSuppressions
 				.Where (suppression => ((DiagnosticId) suppression.SuppressMessageInfo.Id).GetDiagnosticCategory () == DiagnosticCategory.Trimming)
-				.Where (suppression => ((DiagnosticId) suppression.SuppressMessageInfo.Id) != DiagnosticId.RedundantSuppression);
+				.Where (suppression => ((DiagnosticId) suppression.SuppressMessageInfo.Id) != DiagnosticId.RedundantSuppression)
+				.Where (suppression => ProviderIsMarked (suppression.Provider));
 
 			foreach (var suppression in redundantSuppressions) {
 				var source = context.Suppressions.GetSuppressionOrigin (suppression);
 
 				context.LogWarning (new MessageOrigin (source), DiagnosticId.RedundantSuppression, $"IL{suppression.SuppressMessageInfo.Id:0000}");
 			}
+
+
+			bool ProviderIsMarked (ICustomAttributeProvider provider)
```

Comment:
I don't have a strong preference for checking the accessors vs adding the property/event into the Annotations cache when executing MarkProperty/MarkEvent but if somebody has reasons to use the latter I'm leaving this comment to flag it

Reply:
Currently we don't do this, but it's likely we'll start soon: To reduce size, if we can prove the property is not reflection accessed we would remove the property and only leave the accessors as simple methods. So there will be a distinct difference between marking property and marking its accessor (even though currently linker doesn't do this).

But currently we don't "mark" properties (no annotation), so checking the accessors is OK. Could you please add a test where there's a property (and/or event) which is targeted by a suppression and:
* It ends up being removed (no usage)
* Used in code, but not in reflection
* Use by reflection

This should make sure that we catch the problem in this code once we implement the above mentioned optimization.
There's also probably an issue for the optimization, so maybe add a comment there that we'll need to update the logic here.